integration/apidefinition_test.go

```diff
@@ -1093,3 +1095,302 @@ func TestAPIDefinition_GraphQL_ExecutionMode(t *testing.T) {
+
+				// Generate another ApiDefinition CR and create it.
+				api := generateApiDef(testNs, func(definition *v1alpha1.ApiDefinition) {
+					definition.ObjectMeta = metav1.ObjectMeta{
+						Name: "another-api", Namespace: testNs,
+					}
+					definition.Spec.GraphQL = &model.GraphQLConfig{
+						GraphRef: testSubGraphCRMetaName, ExecutionMode: model.SubGraphExecutionMode,
+					}
+				})
+				_, err := util.CreateOrUpdate(ctx, r.Client, api, func() error {
+					return nil
+				})
+				eval.NoErr(err)
+
+				// Reconciliation must fail because the SubGraph is already linked by another ApiDefinition since linking
+				// multiple ApiDefinition to one SubGraph CR is forbidden.
+				err = wait.For(conditions.New(c.Client().Resources()).ResourceMatch(api, func(object k8s.Object) bool {
+					_, err = r.Reconcile(ctx, ctrl.Request{NamespacedName: cr.ObjectKeyFromObject(api)})
+					return errors.Is(err, controllers.ErrMultipleLinkSubGraph)
+				}),
+					wait.WithTimeout(defaultWaitTimeout),
+					wait.WithInterval(defaultWaitInterval),
+				)
+				eval.NoErr(err)
+
+				return ctx
+			}).
+		Assess("update ApiDefinition GraphRef to another SubGraph CR",
+			func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+				eval := is.New(t)
+
+				testNs, ok := ctx.Value(ctxNSKey).(string)
+				eval.True(ok)
+
+				const (
+					newSgName = "new-sg"
+					newSDL    = "newSDL"
+					newSchema = "newSchema"
+				)
+
+				// Generate another SubGraph CR and create it.
+				sg := generateSubGraphCR(testNs, func(subGraph *v1alpha1.SubGraph) {
+					subGraph.ObjectMeta.Name = newSgName
+					subGraph.Spec.SDL = newSDL
+					subGraph.Spec.Schema = newSchema
+				})
+				_, err := util.CreateOrUpdate(ctx, r.Client, sg, func() error {
+					return nil
+				})
+				eval.NoErr(err)
+
+				// Get ApiDefinition and update it based on the new SubGraph CR information.
+				api := &v1alpha1.ApiDefinition{ObjectMeta: metav1.ObjectMeta{Name: testApiDef, Namespace: testNs}}
+				err = wait.For(
+					conditions.New(c.Client().Resources()).ResourceMatch(api, func(object k8s.Object) bool {
+						return true
+					}),
+					wait.WithTimeout(defaultWaitTimeout),
+					wait.WithInterval(defaultWaitInterval),
+				)
+				eval.NoErr(err)
+
+				_, err = util.CreateOrUpdate(ctx, r.Client, api, func() error {
+					api.Spec.GraphQL.GraphRef = newSgName
+					return nil
+				})
+				eval.NoErr(err)
+
+				// Wait for reconciliation; so that, the ApiDefinition is updated according to new SubGraph CR.
+				err = wait.For(
+					conditions.New(c.Client().Resources()).ResourceMatch(api, func(object k8s.Object) bool {
+						_, err = r.Reconcile(ctx, ctrl.Request{NamespacedName: cr.ObjectKeyFromObject(api)})
+						return err == nil
+					}),
+					wait.WithTimeout(defaultWaitTimeout),
+					wait.WithInterval(defaultWaitInterval),
+				)
+				eval.NoErr(err)
+
+				// After successful reconciliation, check that ApiDefinition CR is updated properly.
```

Comment:
Can you also validate if subgraph status is updated correctly?

Reply:
Thanks! I updated accordingly with 7d43c8c17aa5f5a514c37f58dae75d62b451d0d0